packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -369,6 +384,36 @@ class DeviceDomain extends Domain {
     return new Future<Null>.value();
   }
 
+  /// Forward a host port to a device port.
+  Future<Null> forward(Map<String, dynamic> args) async {
+    Device device = await _getDevice(daemon, args);
+
+    if (args['devicePort'] is! int)
+      throw 'devicePort is required';
+    int devicePort = args['devicePort'];
+
+    if (args['hostPort'] is! int)
```

Comment:
If we make the host port optional, then when we do forwarding we can bind to whatever's available, and the bound port returned in the json result.